build/configs/rtl8721csm/rtl8721csm_download.sh

```diff
@@ -18,366 +18,50 @@
-	./amebad_image_tool "download" $TTYDEV 1 ${offsets[$partidx]} ${exe_name}
-
-	echo ""
-	echo "Download $exe_name COMPLETE!"
+	./amebad_image_tool "download" $1 1 $2 $3
 }
 
-download_all()
+function board_erase()
 {
 	cd ${IMG_TOOL_PATH}
-	echo "Starting Download..."
-	found_kernel=false
-
-	for partidx in ${!parts[@]}; do
-
-		if [[ "${CONFIG_APP_BINARY_SEPARATION}" != "y" ]];then
-			if [[ "${parts[$partidx]}" == "userfs" ]];then
-				continue
-			fi
-		fi
-
-		if [[ "${parts[$partidx]}" == "kernel" ]];then
-			if [[ $found_kernel == true ]];then
-				continue
-			fi
-			found_kernel=true
-		fi
-
-		exe_name=$(get_executable_name ${parts[$partidx]})
-		[ "No Binary Match" = "${exe_name}" ] && continue
-
-		echo ""
-		echo "=========================="
-		echo "Downloading ${parts[$partidx]} binary"
-		echo "=========================="
-
-		./amebad_image_tool "download" $TTYDEV 1 ${offsets[$partidx]} ${exe_name}
-	done
-	echo ""
-	echo "Download COMPLETE!"
+	./amebad_image_tool "erase" $1 1 $2 0 $3
 }
 
-erase()
+function post_download()
 {
 	cd ${IMG_TOOL_PATH}
-	echo "Starting Erase..."
-	ota_addr=$(($FLASH_START_ADDR + $CONFIG_AMEBAD_FLASH_CAPACITY))
-	if [[ $2 == "all" || $2 == "ALL" ]];then
-		found_kernel=false
-		flash_ota=false
-		echo ""
-		echo "=========================="
-		echo "      Erasing All"
-		echo "=========================="
-		for partidx in ${!parts[@]}; do
-			if [[ "${parts[$partidx]}" == "userfs" ]];then
-				continue
-			elif [[ "${parts[$partidx]}" == "ss" ]];then
-				continue
-			else
-				echo ""
-				echo "=========================="
-				echo "Erasing ${parts[$partidx]} partition"
-				echo "=========================="
-			fi
-
-			./amebad_image_tool "erase" $TTYDEV 1 ${offsets[$partidx]} 0 ${sizes[partidx]}
-		done
-	else
-		for partidx in ${!parts[@]}; do
-			if [[ $2 == "kernel" || $2 == "KERNEL" ]];then
-				if [[ "${parts[$partidx]}" != "kernel" ]];then
-					continue
-				elif [[ ${offsets[$partidx]} -lt ${ota_addr} ]];then
-					ota_addr=0
-					found_kernel=false
-				else
-					continue
-				fi
-			elif [[ $2 == "ota" || $2 == "OTA" ]];then
-				if [[ "${parts[$partidx]}" != "kernel" ]];then
-					continue
-				elif [[ ${offsets[$partidx]} -lt ${ota_addr} ]];then
-					ota_addr=${offsets[$partidx]}
-					continue
-				else
-					flash_ota=false
-				fi
-			elif [[ $2 == "ss" || $2 == "SS" ]];then
-				if [[ "${parts[$partidx]}" != "ss" ]];then
-					continue
-				fi
-			elif [[ $2 == "userfs" || $2 == "USERFS" ]];then
-				if [[ "${parts[$partidx]}" != "userfs" ]];then
-					continue
-				fi
-			else
-				printf "\n## Invalid Option ##\n\n"
-				rtl8721csm_dwld_help
-				break
-			fi
-
-			echo ""
-			echo "=========================="
-			echo "Erasing ${parts[$partidx]} partition"
-			echo "=========================="
-			./amebad_image_tool "erase" $TTYDEV 1 ${offsets[$partidx]} 0 ${sizes[partidx]}
-		done
+	[ -e km0_boot_all.bin ] && rm km0_boot_all.bin
```

Comment:
If we can read the `binary_name_map.txt` in here, then we can replace the .bin name like BL1, BL2.

Reply:
Done.